Sources/Build/misc.swift

```diff
@@ -32,6 +35,83 @@ extension CModule {
     }
 }
 
+extension ClangModule {
+    
+    public enum ModuleMapError: ErrorProtocol {
+        case UnsupportedIncludeLayoutForModule(String)
+    }
+    
+    public func generateModuleMap() throws {
+        
+        //Return if module map is already present
+        guard !moduleMapPath.isFile else {
```

Comment:
What happens when a new header file is added and the module map file is checked in? Won't that not generate a new modulemap and thus not make the new header file available?